Backendless/Transaction/UnitOfWorkDeleteImpl.cs

```diff
@@ -0,0 +1,137 @@
+    {
+      String objectId = TransactionHelper.ConvertObjectMapToObjectId( objectMap );
+      return Delete( tableName, objectId );
+    }
+
+    public OpResult Delete( String tableName, String objectId )
+    {
+      String operationResultId = opResultIdGenerator.GenerateOpResultId( OperationType.DELETE, tableName );
+      OperationDelete operationDelete = new OperationDelete( OperationType.DELETE, tableName, operationResultId, objectId );
+
+      operations.AddLast( operationDelete );
+      return TransactionHelper.MakeOpResult( tableName, operationResultId, OperationType.DELETE );
+    }
+
+    public OpResult Delete( OpResult result )
+    {
+      if( result == null )
+        throw new ArgumentException( ExceptionMessage.NULL_OP_RESULT );
+
+      if( !OperationTypeUtil.supportEntityDescriptionResultType.Contains( result.GetOperationType() ) )
+        throw new ArgumentException( ExceptionMessage.REF_TYPE_NOT_SUPPORT );
+
+      String operationResultId = opResultIdGenerator.GenerateOpResultId( OperationType.DELETE, result.GetTableName() );
+      OperationDelete operationDelete = new OperationDelete( OperationType.DELETE, result.GetTableName(), operationResultId,
+                                                              result.ResolveTo( "objectId" ).MakeReference() );
+      operations.AddLast( operationDelete );
+
+      return TransactionHelper.MakeOpResult( result.GetTableName(), operationResultId, OperationType.DELETE );
+    }
+
+    public OpResult Delete( OpResultValueReference resultIndex )
+    {
+      if( resultIndex == null )
+        throw new ArgumentException( ExceptionMessage.NULL_OP_RESULT_VALUE_REFERENCE );
+
+      if( resultIndex.GetResultIndex() == null || resultIndex.GetPropName() != null )
+        throw new ArgumentException( ExceptionMessage.OP_RESULT_INDEX_YES_PROP_NAME_NOT );
+
+      Dictionary<String, Object> referenceToObjectId = TransactionHelper.ConvertCreateBulkOrFindResultIndexToObjectId( resultIndex );
+
+      String operationResultId = opResultIdGenerator.GenerateOpResultId( OperationType.DELETE, resultIndex.GetOpResult().GetTableName() );
+      OperationDelete operationDelete = new OperationDelete( OperationType.DELETE, resultIndex.GetOpResult().GetTableName(),
+                                                                                   operationResultId, referenceToObjectId );
+      operations.AddLast( operationDelete );
+
+      return TransactionHelper.MakeOpResult( resultIndex.GetOpResult().GetTableName(), operationResultId, OperationType.DELETE );
+    }
+
+    public OpResult BulkDelete<E>( List<E> instances )
+    {
+      List<Dictionary<String, Object>> serializedEntities = new List<Dictionary<string, object>>();
+
+      for( int i = 0; i < instances.Count; i++ )
+        serializedEntities.Add( new Dictionary<String, Object>( TransactionHelper.ConvertInstanceToMap<E>( instances[ i ] ) ) );
```

Comment:
This looks very wrong. You create a dictionary for each instance. Each dictionary contains map created from instance. On top of this, I do not understand why you need to convert instance to map where in reality instance can be sent to the server